### PR TITLE
Core/Spells: Ice Barrier now shouldn't trigger Shattered Barrier on dispel

### LINKED ISSUE #14555: 3.3.5 Mage: Ice Barrier
Core revision: https://github.com/TrinityCore/TrinityCore/commit/df4b175d689fa99aa3876a2c26bb24ba09970a98
DB version: TDB 335.58
Script version: Mage_ai
Ice Barrier (http://wotlk.openwow.com/spell=43039)
Shattered Barrier (http://wotlk.openwow.com/spell=54787)
ice barrier gets dispelled it activates the Shattered Barrier
Ice Barrier's removal should only freeze surrounding enemies when the shield is broken by damage, not when it is dispelled or spell stolen.

Please excuse my English.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/11210853-3-3-5-mage-ice-barrier?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Goatform**
Fixes #14555

### CODE REVIEW

**Comment by Kittnz** [comment]
Is it posible to explain a bit what is changing here?

**Comment by SmythX** [comment]
Your fix works, thanks!

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 1]
```diff
@@ -1423,12 +1423,17 @@ void Aura::HandleAuraSpecificMods(AuraApplication const* aurApp, Unit* caster, b
                 if (!caster)
                     break;
                 // Ice barrier - dispel/absorb remove
-                if (removeMode == AURA_REMOVE_BY_ENEMY_SPELL && GetSpellInfo()->SpellFamilyFlags[1] & 0x1)
+                if ((removeMode == AURA_REMOVE_BY_ENEMY_SPELL) &&
+                    (GetSpellInfo()->SpellFamilyFlags[EFFECT_1] & 0x00000001) &&
```
Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: cosmetic note - SpellFamilyFlags is not a per-effect array, you should not use EFFECT_* enum to access it (even if the value is the same)

Comment (ccrs) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: How should it be done then?
All other usages of SpellFamilyFlags seem to do the same

Comment (Shauren) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: just use numbers (and no, other usages don't do that)

File: src/server/game/Spells/Auras/SpellAuras.cpp [hunk 2]
```diff
@@ -1423,12 +1423,17 @@ void Aura::HandleAuraSpecificMods(AuraApplication const* aurApp, Unit* caster, b
...
+                    (GetEffect(EFFECT_0)->GetAmount() <= 0))
                 {
                     // Shattered Barrier
-                    if (AuraEffect* dummy = caster->GetDummyAuraEffect(SPELLFAMILY_MAGE, 2945, 0))
-                        if (roll_chance_i(dummy->GetSpellInfo()->ProcChance))
-                            caster->CastSpell(target, 55080, true, NULL, GetEffect(0));
+                    if (AuraEffect * aurEff = target->GetDummyAuraEffect(SPELLFAMILY_MAGE, 2945, EFFECT_0))
```
Comment (Carbenium) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: AuraEffect* aurEff

**Comment by jackpoz** [comment]
@Goatform can you please reply to https://github.com/TrinityCore/TrinityCore/pull/14605#issuecomment-96161182 and update the code according to https://github.com/TrinityCore/TrinityCore/pull/14605#discussion_r31389545 ?

File: src/server/game/Spells/Auras/SpellAuras.cpp (hunk 2 above)
Comment (Goatform) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: @Shauren do I need to replace this EFFECT_0 with 0 as well ?

File: src/server/game/Spells/Auras/SpellAuras.cpp (hunk 1 above)
Comment (Goatform) [inline src/server/game/Spells/Auras/SpellAuras.cpp]: So this function:

SpellFamilyFlags[EFFECT_1] 

Should be replaced with: SpellFamilyFlags[1]

**Comment by Goatform** [comment]
@jackpoz is it okay, now ?

**Comment by jackpoz** [comment]
can you please explain what's happening there with the additional check GetEffect(EFFECT_0)->GetAmount() <= 0) ?

**Comment by robinsch** [comment]
@jackpoz It's the absorb amount from the shield. With this check we make sure that the shield was destroyed by damage and not removed by dispel or the caster itself.

**Comment by Goatform** [comment]
It's check for spells such as dispel magic.

**Comment by Goatform** [comment]
Yeah, exactly what @robinsch said.

**Comment by LuqJensen** [comment]
better to write a spellscript and remove that hack from SpellAuras.cpp

**Pull request closed without merging**